docs/networking/agc-lab.md

```diff
@@ -0,0 +1,762 @@
+az provider register --namespace Microsoft.NetworkFunction
+az provider register --namespace Microsoft.ServiceNetworking
+
+# Install Azure CLI extensions.
+az extension add --name alb
+```
+
+### Step 3: Install Helm CLI
+
+If you don't have Helm installed in your client machine, do so with the following command:
+
+```bash
+curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash
+```
+
+## Expose an application over HTTP
+
+### Install and configure the ALB Controller
+
+The ALB Controller is a Kubernetes deployment that orchestrates configuration and deployment of Application Gateway for Containers. It uses both ARM and configuration APIs to propagate configuration to the Application Gateway for Containers Azure deployment. [Each release of ALB Controller](https://learn.microsoft.com/en-us/azure/application-gateway/for-containers/alb-controller-release-notes) has a documented helm chart version and supported Kubernetes cluster version. After installing it in a cluster, it will apply a set of CRDs to the cluster (as of version 1.7.9):
+
+- ApplicationLoadBalancer
+- BackendLoadBalancingPolicy
+- BackendTLSPolicy
+- FrontendTLSPolicy
+- HealthCheckPolicy
+- IngressExtension
+- RoutePolicy
+- WebApplicationFirewallPolicy
+
+View the architecture of Application Gateway with Containers in the image below:
+
+![Application Gateway for Containers architecture](https://learn.microsoft.com/en-us/azure/application-gateway/for-containers/media/overview/application-gateway-for-containers-kubernetes-conceptual.png)
+
+#### Identity configuration
+
+Create a user managed identity for ALB controller and federate the identity as Workload Identity to use in the AKS cluster.
+
+```bash
+IDENTITY_RESOURCE_NAME='azure-alb-identity'
+
+MC_RG_NAME=$(az aks show --resource-group ${RG_NAME} --name ${AKS_NAME} --query "nodeResourceGroup" -o tsv)
+MC_RG_ID=$(az group show --name ${MC_RG_NAME} --query id -o tsv)
+
+echo "Creating identity $IDENTITY_RESOURCE_NAME in resource group $RG_NAME"
+az identity create --resource-group $RG_NAME --name $IDENTITY_RESOURCE_NAME
+PRINCIPAL_ID=$(az identity show -g $RG_NAME -n $IDENTITY_RESOURCE_NAME --query principalId -otsv)
+
+echo "Waiting 60 seconds to allow for replication of the identity..."
+sleep 60
+
+echo "Apply Reader role to the AKS managed cluster resource group for the newly provisioned identity"
+az role assignment create --assignee-object-id $PRINCIPAL_ID --assignee-principal-type ServicePrincipal --scope $MC_RG_ID --role "acdd72a7-3385-48ef-bd42-f606fba81ae7" # Reader role
+
+echo "Set up federation with AKS OIDC issuer"
+AKS_OIDC_ISSUER="$(az aks show -n $AKS_NAME -g $RG_NAME --query "oidcIssuerProfile.issuerUrl" -o tsv)"
+az identity federated-credential create --name $IDENTITY_RESOURCE_NAME --identity-name "$IDENTITY_RESOURCE_NAME" --resource-group $RG_NAME --issuer "$AKS_OIDC_ISSUER" --subject "system:serviceaccount:azure-alb-system:alb-controller-sa"
```

Comment:
Can we split break this up into multiple code blocks and do a bit more explanation on what the user is running here and why?